### PR TITLE
Fix optimizely for local dev

### PR DESCRIPTION
**Pull request by piazzatron**
### Description

Previous change to default to stage didn't work because we would take the stage eth provider url instead of remote. This change checks which environment we're in, and uses that to determine which source to pull the eth provider from.

### Dragons

Doesn't seem like there are any other instances of remote config overriding remote dev env
*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

Tested against stage + remote
*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

**Review by raymondjacobson — APPROVED** [review]
im coo w/ dis. thx for comment.

wondering if possible to just have a dev optimizely env instead of the staging one? maybe that's not even worth the headache

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/piazz-fix-optimizely-local

File: packages/common/src/services/audius-backend/AudiusBackend.ts [hunk 1]
```diff
@@ -717,8 +717,12 @@ export const audiusBackend = ({
   }
 
   function getEthWeb3Config() {
-    const providerUrls =
-      getRemoteVar(StringKeys.ETH_PROVIDER_URLS) || ethProviderUrls
+    // In a dev env, always ignore the remote var which is inherited from staging
+    const isDevelopment = process.env.REACT_APP_ENVIRONMENT === 'development'
```
Comment (dylanjeffers) [inline packages/common/src/services/audius-backend/AudiusBackend.ts:721]: So with Audius-backend we are trying to make it env agnostic. The pattern is to put this in the env.ts file in common, and pass env into Audius backend. Lmk if I can help here

**Review by dylanjeffers — CHANGES_REQUESTED** [review]
Generally good, but let's get that car into env

File: packages/common/src/services/audius-backend/AudiusBackend.ts (hunk 1 above)
Comment (raymondjacobson) [reply packages/common/src/services/audius-backend/AudiusBackend.ts:721]: ah good call. thx @dylanjeffers . i was not observant enough here

**Comment by piazzatron** [comment]
> im coo w/ dis. thx for comment.
> 
> 
> 
> wondering if possible to just have a dev optimizely env instead of the staging one? maybe that's not even worth the headache

I think too painful to maintain consistent variables between optimizely envs - even keeping stage + prod in sync is annoying :|

File: packages/mobile/src/services/env.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
 export const env = {
-  EAGER_DISCOVERY_NODES: process.env.REACT_APP_EAGER_DISCOVERY_NODES
```
Comment (dylanjeffers) [inline packages/mobile/src/services/env.ts:2]: really close! for mobile, its `Config.BLAH` where BLAH is the thing minus the REACT_APP, so `ENVIRONMENT: Config.ENVIRONMENT` but also don't trip much cause this is all fleshed out in the the native-reloaded branch actually

**Review by dylanjeffers — APPROVED** [review]
nicee

**Pull request merged**